django/core/cache/backends/redis.py

```diff
@@ -0,0 +1,239 @@
+"""Redis cache backend."""
+
+import pickle
+import random
+import re
+import urllib
+
+import redis
+
+from django.core.cache.backends.base import DEFAULT_TIMEOUT, BaseCache
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.functional import cached_property
+from django.utils.module_loading import import_string
+
+
+class BaseCacheSerializer:
+    def dumps(self):
+        raise NotImplementedError
+
+    def loads(self):
+        raise NotImplementedError
+
+
+class PickleCacheSerializer(BaseCacheSerializer):
+    def __init__(self, options):
+        self._pickle_version = -1
+
+        if 'PICKLE_VERSION' in options:
+            try:
+                self._pickle_version = int(options['PICKLE_VERSION'])
+            except (ValueError, TypeError):
+                raise ImproperlyConfigured('must be an integer')
+
+    def dumps(self, value):
+        return pickle.dumps(value, self._pickle_version)
```

Comment:
Rather than passing all of `OPTIONS` here we should only pass the protocol:
```suggestion
    def __init__(self, protocol=None):
        self._protocol = pickle.HIGHEST_PROTOCOL if protocol is None else protocol

    def dumps(self, value):
        return pickle.dumps(value, self._protocol)
```

Reply:
Sure, this is much more clean.